ophyd/signal.py

```diff
@@ -475,12 +483,13 @@ def get(self, *, as_string=None, **kwargs):
         if as_string is None:
             as_string = self._string
 
-        if not self._read_pv.connected:
-            if not self._read_pv.wait_for_connection():
-                raise TimeoutError('Failed to connect to %s' %
-                                   self._read_pv.pvname)
+        with self._lock:
+            if not self._read_pv.connected:
+                if not self._read_pv.wait_for_connection():
```

Comment:
Should the `timeout` be propagated here as well. Perhaps it would be better to make it configurable via a separate kwargs (`configuration_timeout`)?